### PR TITLE
Fix/delayed terminate

### PR DESCRIPTION
**Pull request by lafirest**

### CODE REVIEW

File: lib-ce/emqx_modules/src/emqx_mod_delayed.erl [hunk 1]
```diff
@@ -49,15 +49,19 @@
         , code_change/3
         ]).
 
--record(delayed_message,
-        { key
-        , msg
-        }).
+-record(delayed_message, {key, msg}).
```
Comment (zmstone) [inline lib-ce/emqx_modules/src/emqx_mod_delayed.erl:52]: 风格没啥问题的话尽量别改

File: lib-ce/emqx_modules/src/emqx_mod_delayed.erl [hunk 2]
```diff
@@ -163,8 +169,9 @@ handle_info(Info, State) ->
     ?LOG(error, "Unexpected info: ~p", [Info]),
     {noreply, State}.
 
-terminate(_Reason, #{timer := TRef}) ->
-    emqx_misc:cancel_timer(TRef).
+terminate(_Reason, #{timer := PublishTimer, stats_timer := StatsTimer}) ->
```
Comment (zmstone) [inline lib-ce/emqx_modules/src/emqx_mod_delayed.erl:172]: Backward compatibility: the stats timer is newly added, may not exist after relup

File: lib-ce/emqx_modules/src/emqx_mod_delayed.erl [hunk 3]
```diff
@@ -179,12 +186,14 @@ code_change(Vsn, State, _Extra) when Vsn =:= "4.3.0" ->
 %%--------------------------------------------------------------------
 
 %% Ensure the stats
+-spec ensure_stats_event(state()) -> state().
 ensure_stats_event(State) ->
     StatsFun = emqx_stats:statsfun('delayed.count', 'delayed.max'),
     {ok, StatsTimer} = timer:send_interval(timer:seconds(1), stats),
-    State#{stats_fun => StatsFun, stats_timer => StatsTimer}.
+    State#{stats_fun => StatsFun, stats_timer := StatsTimer}.
```
Comment (zmstone) [inline lib-ce/emqx_modules/src/emqx_mod_delayed.erl:193]: Is it safe to use := after relup?

**Review by HJianBo — APPROVED** [review]

**Comment by HJianBo** [comment]
plz update `CHANGE-4.3.md`

**Pull request merged**